stdlib/public/Concurrency/Task.cpp

```diff
@@ -1671,7 +1672,8 @@ static void swift_task_asyncMainDrainQueueImpl() {
   }
 #endif
 
-    dispatch_main();
+  dispatch_main();
+  swift_Concurrency_fatalError("Returned from dispatch_main()");
```

Comment:
Why not just make this `swift_unreachable`?

Reply:
That is what I did before but you said that did not make sense.